src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationOptions.cs

```diff
@@ -233,6 +234,11 @@ public OpenIdConnectProtocolValidator ProtocolValidator
         public bool RefreshOnIssuerKeyNotFound { get; set; }
 
         /// <summary>
+        /// Determines whether authorization requests should be made using POST.
+        /// </summary>
+        public bool UsePostAuthorizationRequests { get; set; }
```

Comment:
Done :smile: 

Allowing the developer to change the authentication method directly from the `RedirectToIdentityProvider` notification would be nice, but sadly, `RedirectToIdentityProviderNotification` is not OIDC-specific and is part of `Microsoft.AspNet.Authentication`.

To work around that, we could create a derived class or add a "generic" property to `RedirectToIdentityProvider` containing the HTTP method.